source/scripting/Entity.cs

```diff
@@ -561,6 +561,22 @@ public bool HasBeenDamagedBy(Entity entity)
 		{
 			return Function.Call<bool>(Hash.HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY, Handle, entity.Handle, 1);
 		}
+		public bool HasBeenDamagedBy(WeaponHash weapon)
```

Comment:
IMO this should be called `HasBeenDamagedByWeapon` since there's no other overloads.